control/grpc.py

```diff
@@ -35,17 +36,45 @@ class GatewayService(pb2_grpc.GatewayServicer):
         spdk_rpc_client: Client of SPDK RPC server
     """
 
-    def __init__(self, config, gateway_state, spdk_rpc_client) -> None:
+    def __init__(self, config, gateway_state, omap_state, spdk_rpc_client) -> None:
         """Constructor"""
         self.logger = logging.getLogger(__name__)
         self.config = config
         self.gateway_state = gateway_state
+        self.omap_state = omap_state
         self.spdk_rpc_client = spdk_rpc_client
         self.gateway_name = self.config.get("gateway", "name")
         if not self.gateway_name:
             self.gateway_name = socket.gethostname()
+        self.omap_file_lock_duration = self.config.getint_with_default("gateway", "omap_file_lock_duration", 60)
+        self.omap_file_update_retries = self.config.getint_with_default("gateway", "omap_file_update_retries", 10)
+        self.enter_args = {}
         self._init_cluster_context()
 
+    def __call__(self, **kwargs):
```

Comment:
@gbregman propose to extract context manager into a separate class, see [here](https://github.com/baum/ceph-nvmeof/commit/1f531ad3aac87ec06362d9de0d58a8d63623275b). WDYT?

Reply:
@baum I don't really see the need for that.